h/views/api/auth.py

```diff
@@ -104,6 +104,7 @@ def post(self):
         request_param="response_mode=web_message",
         is_authenticated=True,
         renderer="h:templates/oauth/authorize_web_message.html.jinja2",
+        require_csrf=False,
```

Comment:
FIXME: This is needed to get some functests passing. This line should be removed and the failing functests should be fixed to use a CSRF token.

Reply:
Can you add a comment here to indicate that this is being added only for the tests, and should not be required in actual usage.

Reply:
It turned out the failing functests were failing because the app was actually broken: the OAuth-related form in the "Hypothesis is requesting access to your Hypothesis account" popup window when you log in to the client was missing a CSRF token and was actually failing due to this PR's newly-added CSRF protection. Added a hidden input field with a CSRF token to that form, and removed ths `require_csrf=False` here.